controllers/workspace/devworkspace_controller.go

```diff
@@ -116,6 +119,18 @@ func (r *DevWorkspaceReconciler) Reconcile(req ctrl.Request) (reconcileResult ct
 		return reconcile.Result{Requeue: true}, err
 	}
 
+	// Stop failed workspaces
+	if workspace.Status.Phase == dw.DevWorkspaceStatusFailed && workspace.Spec.Started {
+		patch := []byte(`{"spec":{"started": false}}`)
+		err := r.Client.Patch(context.Background(), workspace, client.RawPatch(types.StrategicMergePatchType, patch))
```

Comment:
Strategic merge patch is not supported for CRs. 

```suggestion
		err := r.Client.Patch(context.Background(), workspace, client.RawPatch(types.MergePatchType, patch))
```

Reply:
Thanks for pointing this out @amisevsk! I see it's not supported in the docs here: https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/